### ISSUE TITLE
Error with Style/FileWrite

### ISSUE THREAD

**Issue by jrmhaig** [rubocop/rubocop#15260]
### Before submitting

- [x] I have searched the existing issues and this hasn't been reported yet.
- [x] I am running the latest stable version of RuboCop.

### Issue type

Crash/error

### Cop name

Style/FileWrite

### Steps to reproduce the problem

1. Create a file, `test.rb` with the content;

```ruby
File.open('/tmp/test.txt', 'wb') { |f| f.write('hello') }
```

2. Execute `bundle exec rubocop test.rb`

### Expected behavior

An offence related to the cop Style/FileWrite is reported.

### Actual behavior

```
Inspecting 1 file
An error occurred while Style/FileWrite cop was inspecting /.../test.rb:3:0.
To see the complete backtrace run rubocop -d.
.

1 file inspected, no offenses detected

1 error occurred:
An error occurred while Style/FileWrite cop was inspecting /.../test.rb:3:0.
Errors are usually caused by RuboCop bugs.
Please, update to the latest RuboCop version if not already in use, and report a bug if the issue still occurs on this version.
https://github.com/rubocop/rubocop/issues

Mention the following information in the issue report:
1.87.0 (using Parser 3.3.11.1, Prism 1.9.0, rubocop-ast 1.49.1, analyzing as Ruby 3.4, running on ruby 3.4.7) [arm64-darwin25]
```

### RuboCop configuration

```yaml
AllCops:
  NewCops: enable
```

### RuboCop version

```shell
% bundle exec rubocop -V
1.87.0 (using Parser 3.3.11.1, Prism 1.9.0, rubocop-ast 1.49.1, analyzing as Ruby 3.4, running on ruby 3.4.7) [arm64-darwin25]
```
